[Part 5] All numeric CSSPrimitiveValue resolvers need to take CSSToLengthConversionData: font

weinig opened the pull request:
#### ea0e8115cda32876124dc327f488b465e7041b0d
<pre>
[Part 5] All numeric CSSPrimitiveValue resolvers need to take CSSToLengthConversionData: font
<a href="https://bugs.webkit.org/show_bug.cgi?id=279677">https://bugs.webkit.org/show_bug.cgi?id=279677</a>

Reviewed by Antti Koivisto and Darin Adler.

Refactors all font related style building into StyleResolveForFont.h/cpp
with &quot;Deprecated&quot; overloads of resolvers that don&apos;t take conversion data
to make it abundantly clear they won&apos;t work correctly in all cases.

Updates Style::BuilderConverter to call the new resolvers, ensuring that
all the converter functions now take Style::BuilderState so they have
access to the conversion data / style data.

Fixes a small issue with font-weight computed value serialization by
ensuring that &quot;oblique&quot; gets serialized for computed value even if
the angle is 0 (previously we would have serialized this as &quot;normal&quot;).

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-invalid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-valid.html:
    - Update tests with cases for calc() and calc() using relative length units.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
    - Update for renamed file.
* Source/WebCore/css/CSSFontFace.cpp:
* Source/WebCore/css/FontSelectionValueInlines.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
* Source/WebCore/html/canvas/CanvasRenderingContext2D.cpp:
* Source/WebCore/html/canvas/OffscreenCanvasRenderingContext2D.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleResolveForFont.cpp: Renamed from Source/WebCore/style/StyleResolveForFontRaw.cpp.
* Source/WebCore/style/StyleResolveForFont.h: Renamed from Source/WebCore/style/StyleResolveForFontRaw.h.

Canonical link: <a href="https://commits.webkit.org/283851@main">https://commits.webkit.org/283851@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/571ba8de17c46f16eb6646cf2261cfac1b130673

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/67598 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/46977 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/20230 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/71647 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/18735 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/69716 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/54775 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/18539 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/54138 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/12531 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/70665 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/43111 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/58458 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/34607 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/39784 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/15863 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/17093 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/61751 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/16204 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/73347 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/11557 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/15526 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/61581 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/11592 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/58527 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/61634 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14974 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/9438 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/3050 "Passed tests") | | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/42783 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/43859 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/45045 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/43600 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/style/StyleResolveForFont.cpp:
```diff
@@ -0,0 +1,537 @@
+        return FontSelectionValue(clampTo<float>(primitiveValue.resolveAsNumberDeprecated(), 1, 1000));
+
+    ASSERT(primitiveValue.isValueID());
+    auto valueID = primitiveValue.valueID();
```
No need for local variable?

weinig committed:
[Part 5] All numeric CSSPrimitiveValue resolvers need to take CSSToLengthConversionData: font
https://bugs.webkit.org/show_bug.cgi?id=279677

Reviewed by Antti Koivisto and Darin Adler.

Refactors all font related style building into StyleResolveForFont.h/cpp
with "Deprecated" overloads of resolvers that don't take conversion data
to make it abundantly clear they won't work correctly in all cases.

Updates Style::BuilderConverter to call the new resolvers, ensuring that
all the converter functions now take Style::BuilderState so they have
access to the conversion data / style data.

Fixes a small issue with font-weight computed value serialization by
ensuring that "oblique" gets serialized for computed value even if
the angle is 0 (previously we would have serialized this as "normal").

* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-size-adjust-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-invalid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-stretch-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-style-valid.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-computed.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-fonts/parsing/font-weight-valid.html:
    - Update tests with cases for calc() and calc() using relative length units.

* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
    - Update for renamed file.
* Source/WebCore/css/CSSFontFace.cpp:
* Source/WebCore/css/FontSelectionValueInlines.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Font.h:
* Source/WebCore/html/canvas/CanvasRenderingContext2D.cpp:
* Source/WebCore/html/canvas/OffscreenCanvasRenderingContext2D.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleResolveForFont.cpp: Renamed from Source/WebCore/style/StyleResolveForFontRaw.cpp.
* Source/WebCore/style/StyleResolveForFont.h: Renamed from Source/WebCore/style/StyleResolveForFontRaw.h.

Canonical link: https://commits.webkit.org/283851@main
